doc/mgmt/Management Framework.md

```diff
@@ -1922,7 +1925,87 @@ The table-transformer is inherited to all descendant nodes unless another one is
 11. `sonic-ext:get-validate [function]`: 
 A special hook to validate YANG nodes, to populate data read from database, allows developers to instruct Transformer to choose a YANG node among multiple nodes, while constructing the response payload. 
 Typically used to check the “when” condition to validate YANG node among multiple nodes to choose only valid nodes from sibling nodes.
+12. `sonic-ext:get-namespace [function]`:
+Multiple DB extension, populates associated namespaces with the YANG path
+Used by Translib service to get all DB's associated with the YANG path. In case of single DB, this extension shall not be used.
+Namespace yang extension shall allow override capability to allow product specific customizations 
+```Sample annotation
+get-namespace annotation sample for openconfig-optical-amplifier yang support: 
+
+Annotation YANG File:
+sonic-mgmt-common/models/yang/annotations/openconfig-optical-amplifier-annot.yang
+module openconfig-optical-amplifier-annot {
+  .....
+    deviation /oc-opt-amp:optical-amplifier/oc-opt-amp:amplifiers/oc-opt-amp:amplifier {
+      deviate add {
+        sonic-ext:key-transformer "oc_name_key_xfmr";
+        sonic-ext:get-namespace "oc_name_get_namespace_xfmr";
+      }
+    }
+  .....
+    deviation /oc-opt-amp:optical-amplifier/oc-opt-amp:amplifiers/oc-opt-amp:amplifier/oc-opt-amp:state {
+      deviate add {
+        sonic-ext:db-name "STATE_DB";
```

Comment:
This example has DB-name annotation , what is the role of namespace in such cases. Does DB-name override the DBs returned from namespace callback or vice-versa?  I guess both DB-name annotation and namespace callback shouldn't co-exist

Reply:
> This example has DB-name annotation , what is the role of namespace in such cases. Does DB-name override the DBs returned from namespace callback or vice-versa? I guess both DB-name annotation and namespace callback shouldn't co-exist

The HLD is to support multiple RedisDB instances in case of a chassis, where separate RedisDB instance shall be present for each slot. Namespace call back will be used to select the RedisDB instance. Each RedisDB instace will contain all existing SONIC DBs like Config_DB, State_DB etc.., 
Now in this case we need two level of selection 
1. Namespace : like ASIC0, ASIC1
    Newly introduced Namespace CB shall be used to select the RedisDB instance
2. DB/Table name: Config_DB (TableA) 
    Existing Annotation will be continued to select the DB/Table in the selected RedisDB instance
Newly introduced Namespace annotation will co-exist with the existing DB-name, Table annotation